examples/templates/envbox/main.tf

```diff
@@ -0,0 +1,305 @@
+}
+
+variable "create_tun" {
+  type        = bool
+  sensitive   = true
+  description = "Add a TUN device to the workspace."
+}
+
+variable "create_fuse" {
+  type        = bool
+  description = "Add a FUSE device to the workspace."
+  sensitive   = true
+}
+
+variable "max_cpus" {
+  type        = string
+  sensitive   = true
+  description = "Max number of CPUs the workspace may use (e.g. 2)."
+}
+
+variable "min_cpus" {
+  type        = string
+  sensitive   = true
+  description = "Minimum number of CPUs the workspace may use (e.g. .1)."
+}
+
+variable "max_memory" {
+  type        = string
+  description = "Maximum amount of memory to allocate the workspace (in GB)."
+  sensitive   = true
+}
+
+variable "min_memory" {
+  type        = string
+  description = "Minimum amount of memory to allocate the workspace (in GB)."
+  sensitive   = true
+}
+
+provider "kubernetes" {
+  # Authenticate via ~/.kube/config or a Coder-specific ServiceAccount, depending on admin preferences
+  config_path = var.use_kubeconfig == true ? "~/.kube/config" : null
+}
+
+data "coder_workspace" "me" {}
+
+resource "coder_agent" "main" {
+  os             = "linux"
+  arch           = "amd64"
+  startup_script = <<EOT
+    #!/bin/bash
+    # home folder can be empty, so copying default bash settings
+    if [ ! -f ~/.profile ]; then
+      cp /etc/skel/.profile $HOME
+    fi
+    if [ ! -f ~/.bashrc ]; then
+      cp /etc/skel/.bashrc $HOME
+    fi
+    # install and start code-server
+    curl -fsSL https://code-server.dev/install.sh | sh -s -- --version 4.8.3 | tee code-server-install.log
+    code-server --auth none --port 13337 | tee code-server-install.log &
+  EOT
+}
+
+# code-server
+resource "coder_app" "code-server" {
+  agent_id     = coder_agent.main.id
+  slug         = "code-server"
+  display_name = "code-server"
+  icon         = "/icon/code.svg"
+  url          = "http://localhost:13337?folder=/home/coder"
+  subdomain    = false
+  share        = "owner"
+
+  healthcheck {
+    url       = "http://localhost:13337/healthz"
+    interval  = 3
+    threshold = 10
+  }
+}
+
+resource "kubernetes_persistent_volume_claim" "home" {
+  metadata {
+    name      = "coder-${lower(data.coder_workspace.me.owner)}-${lower(data.coder_workspace.me.name)}-home"
+    namespace = var.namespace
+  }
+  wait_until_bound = false
+  spec {
+    access_modes = ["ReadWriteOnce"]
+    resources {
+      requests = {
+        storage = "${data.coder_parameter.home_disk.value}Gi"
+      }
+    }
+  }
+}
+
+resource "kubernetes_pod" "main" {
+  count = data.coder_workspace.me.start_count
+  metadata {
+    name      = "coder-${lower(data.coder_workspace.me.owner)}-${lower(data.coder_workspace.me.name)}"
+    namespace = var.namespace
+  }
+  spec {
+    node_selector = {
+      sreya-test = true
+    }
+    container {
+      name              = "dev"
+      image             = "gcr.io/coder-dogfood/sreya/envbox:testtt"
```

Comment:
Can we use a more official image here before we publish?

Reply:
Lol for sure, I intended to update it